ListApp/Tests/ListApp.Api.Tests/ItemControllerTests.cs

```diff
@@ -21,17 +25,44 @@ public class ItemsControllerTests
         private static readonly ListItem PostedItem = new ListItem { Id = PostedItemGuid, Text = PostedItemText };
 
         private ItemsController _itemsController;
+        private IRepository _itemsRepository;
+        private IRouteHelper _routeHelper;
 
         [SetUp]
         public void SetUp()
         {
-            _itemsController = new ItemsController();
-            _itemsController.Configuration = Substitute.For<HttpConfiguration>();
-            _itemsController.Request = Substitute.For<HttpRequestMessage>();
+            _itemsRepository = Substitute.For<IRepository>();
+            _itemsRepository.GetAllAsync()
+                .Returns(Constants.MockListItems);
+            _itemsRepository.GetAsync(Arg.Any<Guid>())
+                .Returns(Constants.MockListItems.ElementAt(0));
+            _itemsRepository.AddAsync(Arg.Any<ListItem>())
+                .Returns(Constants.MockListItems.ElementAt(0));
+            _itemsRepository.DeleteAsync(Arg.Any<Guid>())
+                .Returns(Constants.MockListItems.ElementAt(0));
+            _itemsRepository.UpdateAsync(Arg.Any<Guid>(), Arg.Any<ListItem>())
+                .Returns(Constants.MockListItems.ElementAt(0));
```

Comment:
You don't need to mock all of the methods here. The test methods can mock the method(s) they need with their own expected results.